apis/apps/v1alpha1/broadcastjob_types.go

```diff
@@ -153,9 +153,10 @@ const (
 // FailurePolicy indicates the behavior of the job, when failed pod is found.
 type FailurePolicy struct {
 	// Type indicates the type of FailurePolicyType.
+	// Default is FailurePolicyTypeFailFast.
 	Type FailurePolicyType `json:"type,omitempty" protobuf:"bytes,1,opt,name=type,casttype=FailurePolicyType"`
 
-	// RestartLimit specifies the number of retries before marking the pod failed.
+	// RestartLimit specifies the number of retries before marking the job failed.
```

Comment:
That is the failed pod, not the whole job.

Reply:
The pod is the worker, when the pod failed, it will be deleted and the job will create another new pod to retry.
If the pod failed times > `RestartLimit`, the parent job will be marked failed.

Is something missing or misunderstood?

Reply:
No, broadcastjob will not recreate the pod when it fails.

It only compares `restartLimit` with the `restartCount` of the pod status. When a pod has a `restartCount` bigger than `restartLimit`, it will be considered as a failed pod.

https://github.com/openkruise/kruise/blob/286792678707b67bc9abff2a342195ececd3d8f2/pkg/controller/broadcastjob/broadcastjob_utils.go#L66-L85

Reply:
After checking the code and verify the demo yaml in my cluster, we can see the pod phase will be `Failed` and deleted immediately, not by calculating `RestartCount` in status.

The logic to get failed pod is here:
https://github.com/openkruise/kruise/blob/286792678707b67bc9abff2a342195ececd3d8f2/pkg/controller/broadcastjob/broadcastjob_utils.go#L44-L64

The logic goes into L50-L51: checking the pod phase is `Failed`, then the pod was added into failedPods, then
will go to the delete pod logic on code L282 and L294:
https://github.com/openkruise/kruise/blob/286792678707b67bc9abff2a342195ececd3d8f2/pkg/controller/broadcastjob/broadcastjob_controller.go#L273-L297

Please check the logic again, thanks.

Reply:
Yeah, so in this doc it should be `pod` instead of `job`.

Reply:
OK, updated. PTAL thanks.
Also updated the website doc: https://github.com/openkruise/openkruise.io/pull/68